docs: update the version of the action used in the README example (#977)
golangci/golangci-lint-action: README.md

## Compatibility

> * `v3.0.0+` requires explicit setup-go installation step prior to using this action: `uses: actions/setup-go@v5`.

Reviewer: ```suggestion
* `v4.0.0+` requires an explicit setup-go installation step before using this action: `uses: actions/setup-go@v5`.
```
